backend/app/api/endpoints/groups.py

```
@@ -69,6 +69,32 @@ def list_groups(
     return GroupListResponse(total=total, items=groups)
 
 
+@router.get("/search", response_model=GroupListResponse)
+def search_groups_endpoint(
+    q: str = Query("", min_length=0, max_length=100, description="Search query for group name or display_name"),
+    page: int = Query(1, ge=1, description="Page number"),
+    limit: int = Query(20, ge=1, le=100, description="Items per page"),
+    db: Session = Depends(get_db),
```

Ruff (0.15.12)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)